src/Tools/Common/Commands/ProcessStatus.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.CommandLine;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+using Microsoft.Diagnostics.Tools.RuntimeClient;
+
+namespace Microsoft.Internal.Common.Commands
+{
+    public class ProcessStatusCommandHandler
+    {
+        /// <summary>
+        /// Print the current list of available .NET core processes for diagnosis and their statuses
+        /// </summary>
+        public static void PrintProcessStatus(IConsole console)
+        {
+            try
+            {
+                StringBuilder sb = new StringBuilder();
+                var processes = EventPipeClient.ListAvailablePorts()
+                    .Select(GetProcessById)
+                    .Where(process => process != null)
+                    .OrderBy(process => process.ProcessName)
+                    .ThenBy(process => process.Id);
+
+                foreach (var process in processes)
+                {
+                    try
+                    {
+                        sb.Append($"{process.Id, 10} {process.ProcessName, -10} {process.MainModule.FileName}\n");
+                    }
+                    catch (Exception)
```

Comment:
Is there a narrower exception we can catch?

I know this was pre-existing and you are just refactoring, so if this isn't quick to deal with opening an issue to deal with this later is fine too.